sql/sql_expression_operations.cc

```diff
@@ -1369,8 +1369,13 @@ bindUserFunction(SqlBindingScope & context) const
          }
          else
          {
-            throw HttpReturnException(400, "User function " + functionName
-                                   + " expect a row argument ({ }), got " + args[0]->print() );
+            auto functionresult = std::dynamic_pointer_cast<FunctionCallWrapper>(args[0]);
```

Comment:
This smells bad.  The requirement is on the return type (that the expression should return a row), not on the structure of the expression tree.   Why can't we simply test for the output type of the expression rather than checking its actual node?